ir/type.def

```diff
@@ -458,6 +458,10 @@ class Type_MethodCall : Type {
 
 /// Actions look a lot like methods in many respects.
 /// However, invoking an action returns another action
+/// \TODO(mbudiu): where is the constraint of returning an action captured?
```

Comment:
That is part of the type-checker.
The methods may be the same, but the pre/postorder methods will be different.
Also, look at methodInstance.h, in particular the class ActionCall.